tests/OAuth2SummitSponsorApiTest.php

```diff
@@ -356,4 +357,176 @@ public function testAddSponsorUserMember(){
+            'label' => 'Added extra question type',
+            'mandatory' => true,
+        ];
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+        $response = $this->action(
+            "POST",
+            "OAuth2SummitSponsorApiController@addExtraQuestion",
+            $params,
+            [],
+            [],
+            [],
+            $headers,
+            json_encode($data)
+        );
+
+        $content = $response->getContent();
+        $this->assertResponseStatus(201);
+        $question = json_decode($content);
+        $this->assertTrue(!is_null($question));
+        $this->assertTrue($question->name === $name);
+        return $question;
+    }
+
+    public function testGetAllSponsorExtraQuestionsBySponsor(){
+        $question = $this->testAddSponsorExtraQuestions();
+
+        $params = [
+            'id' => self::$summit->getId(),
+            'sponsor_id' => self::$sponsors[0]->getId(),
+            'filter'=> 'label=='.$question->label,
+            'order' => '+order',
+        ];
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+        $response = $this->action(
+            "GET",
+            "OAuth2SummitSponsorApiController@getExtraQuestions",
+            $params,
+            [],
+            [],
+            [],
+            $headers
+        );
+
+        $content = $response->getContent();
+        $this->assertResponseStatus(200);
+        $page = json_decode($content);
+        $this->assertTrue(!is_null($page));
+        $this->assertNotEmpty($page->data);
+        return $page;
+    }
+
+    public function testGetSponsorExtraQuestionsBySponsor(){
+        $q = $this->testAddSponsorExtraQuestions();
+
+        $params = [
+            'id' => self::$summit->getId(),
+            'sponsor_id' => self::$sponsors[0]->getId(),
+            'extra_question_id' => $q->id
+        ];
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+        $response = $this->action(
+            "GET",
+            "OAuth2SummitSponsorApiController@getExtraQuestion",
+            $params,
+            [],
+            [],
+            [],
+            $headers
+        );
+
+        $content = $response->getContent();
+        $this->assertResponseStatus(200);
+        $question = json_decode($content);
+        $this->assertEquals($q->id, $question->id);
+        return $question;
+    }
+
+    public function testUpdateSponsorExtraQuestionsBySponsor(){
+        $q = $this->testAddSponsorExtraQuestions();
+
+        $params = [
+            'id' => self::$summit->getId(),
+            'sponsor_id' => self::$sponsors[0]->getId(),
+            'extra_question_id' => $q->id
```

Comment:
this test is not complete we need to test ordering update https://github.com/OpenStackweb/summit-api/pull/233/files#r1484347818